src/com/company/Main.java

```diff
@@ -33,44 +55,34 @@ public static String readFileAsString(String fileName)
               } catch (IOException e) {
                   e.printStackTrace();
               }
-              return data;
+              return data.replaceAll("[^a-zA-Z0-9@]", " ");
+
     }
 
-        public static void checkForSupport(File file)
+        public static boolean isFileFormatSupported(File file)
         {
-
-            if (file.getName().endsWith(".txt")) {
-                System.out.println("file supported");
-
-            }
-
-            else
-                {
-                    System.out.println("file not supported");
-                }
+            return (file.getName().endsWith(".txt"));
         }
 
-        public static void checkForExistance(File file) {
-            if (file.exists()) {
-
-                System.out.println("file exists");
-            }
-            else {
-                System.out.println("file doesn't found ");
-            }
-        }
-        public static void avoidSpecialChar(String data)
-        {
-            data = data.replaceAll("[^a-zA-Z0-9@]", " ");
+        public static boolean isFileExists(File file) {
+           return file.exists();
         }
+
             public static void searchTheWord(String data,String searchword) {
                 StringTokenizer st = new StringTokenizer(data);
                 int count = 0;
+                if(st.hasMoreElements()==false)
```

Comment:
check what does st.hasMoreElements() do

Reply:
it was an extra code and i have removed that